cerbero/utils/git.py

```diff
@@ -290,7 +290,7 @@ async def local_checkout(git_dir, local_git_dir, commit, logfile=None):
     # missing
     branch_name = 'cerbero_build'
     shell.call('%s reset --hard %s' % (GIT, commit), local_git_dir, logfile=logfile)
-    shell.call('%s branch %s' % (GIT, branch_name), local_git_dir, fail=False, logfile=logfile)
+    shell.call('%s branch %s %s' % (GIT, branch_name, commit), local_git_dir, fail=False, logfile=logfile)
```

Comment:
I need to double check if something else may be affected. But this is needed because fetch will end up with HEAD pointing to the commit of the fetched recipe, and in the extract step, other recipes specifying other commits, will clone pointing to the wrong commit (the commit of the fetched source).

Reply:
@cfoch I've taken a look at this and I've found two things:
1. We don't need to pass the commit when creating the branch because the previous `reset --hard` already makes sure we're pointing to the right commit
2. We have duplicated calls (from when I move to async). We can consolidate them and do the same thing the community cerbero is doing: https://gitlab.freedesktop.org/gstreamer/cerbero/-/blob/master/cerbero/utils/git.py#L223

```
    await shell.async_call([GIT, 'checkout', commit, '-B', branch_name], local_git_dir, logfile=logfile)
    await shell.async_call([GIT, 'clone', local_git_dir, '-s', '-b', branch_name, '.'],
               git_dir, logfile=logfile)
```

Please change this, rebase with autosquash and we're good to merge. Afterwards, you'll need to update https://github.com/fluendo/fluendo-cerbero/pull/1029 to the new cerbero commit (the merge commit), and then merge the fluendo-cerbero one.